src/test/rustdoc-ui/coverage/allow_missing_docs.rs

```diff
@@ -0,0 +1,39 @@
+// compile-flags:-Z unstable-options --show-coverage
+// check-pass
+
+//! Make sure to have some docs on your crate root
+
+#[allow(missing_docs)]
+pub mod mod_foo {
+    pub struct Bar;
+}
+
+/// This is a struct with a `#[allow(missing_docs)]`
+pub struct AllowTheMissingDocs {
+    #[allow(missing_docs)]
+    pub empty_str: String,
+
+    /// This has
+    #[allow(missing_docs)]
+    /// but also has documentation comments
+    pub hello: usize,
+
+    /// The doc id just to create a boilerplate comment
+    pub doc_id: Vec<u8>,
+}
+
+/// A function that has a documentation
+pub fn this_is_func() {}
+
+#[allow(missing_docs)]
+pub struct DemoStruct {
+    something: usize,
+}
+
+#[allow(missing_docs)]
+pub mod bar {
+    #[warn(missing_docs)]
+    pub struct Bar { //~ WARN
+        pub f: u32, //~ WARN
+    }
```

Comment:
Can you also give an example where the `allow` propogates to an inner item?
```suggestion
    }
    
    pub struct NeedsNoDocs;
```

Reply:
I did it with `DemoStruct` (just above). But I can add another one if you want here?

Reply:
In demo struct it doesn't propagate, the attribute is directly on the item. I want a case where the attribute is on a _parent_, not on the item directly.

Reply:
Fine by me!